nifi-registry-core/nifi-registry-framework/src/main/java/org/apache/nifi/registry/security/authorization/AbstractPolicyBasedAuthorizer.java

```diff
@@ -269,10 +272,13 @@ public final synchronized User updateUser(final User user) throws AuthorizationA
     /**
      * Deletes the user with the given id.
      *
+     * @deprecated use {@link #deleteUser(User)} instead
+     *
      * @param userIdentifier the identifier of the user to delete
      * @return the user that was deleted, or null if no matching user was found
      * @throws AuthorizationAccessException if there was an unexpected error performing the operation
      */
+    @Deprecated
```

Comment:
That makes sense, thanks for the explanation. I guess I would lean towards either of these options, but don't have a preference between the two:

- Remove the deprecated notice, the methods will just be unused
- Remove them all together in this PR since it is a major version anyway